### PR TITLE
docs: Events in README

### PR DESCRIPTION
**Pull request by fabriziodemaria**
WAIT ON RELEASE: #88

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,15 +42,6 @@ The provider is instantiated using a client token that is configured in the Conf
 management API. After that all interaction with the feature flags happens using the OpenFeature client APIs. 
 
 ```java
-package com.spotify.confidence.openfeature;
-
-import com.spotify.confidence.ConfidenceFeatureProvider;
-import dev.openfeature.sdk.Client;
-import dev.openfeature.sdk.MutableContext;
-import dev.openfeature.sdk.OpenFeatureAPI;
-import dev.openfeature.sdk.Value;
```
Comment (nicklasl) [inline README.md:51]: wdyt about keeping this but replacing with a `*` import?
```
import dev.openfeature.sdk.*
```

**Review by nickybondarenko — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -69,3 +60,44 @@ public final class ResolveFlags {
...
+}
+```
+The `send()` API currently supports:
+- Setting a custom event's payload with all the Confidence-supported types via the `ConfidenceValue` interfaces
+- Automatically including the OpenFeature's `Evaluation Context` detected at the time `send()` is called
+  - _Note: this only considers the global Evaluation Context set at the OpenFeatureAPI level_
+- Setting a shared `Event Context` that is going to be appended to each event:
+
+```java
+    final Confidence confidenceWithContext = Confidence.withContext(ConfidenceValue.of("context-value"));
```
Comment (nicklasl) [inline README.md:100]: `Confidence.withContext()` is not a static function.
```suggestion
    final Confidence confidenceWithContext = confidence.withContext(ConfidenceValue.of("context-value"));
```

File: README.md [hunk 3]
```diff
@@ -69,3 +60,44 @@ public final class ResolveFlags {
...
+    // Additionally, events can be emitted
+    confidence.send("my-event", ConfidenceValue.of("event-value"));
+  }
+}
+```
+The `send()` API currently supports:
+- Setting a custom event's payload with all the Confidence-supported types via the `ConfidenceValue` interfaces
+- Automatically including the OpenFeature's `Evaluation Context` detected at the time `send()` is called
+  - _Note: this only considers the global Evaluation Context set at the OpenFeatureAPI level_
+- Setting a shared `Event Context` that is going to be appended to each event:
```
Comment (nicklasl) [inline README.md:97]: This stuff warrants it's own heading.
```suggestion
### Setting a shared `Event Context` that is going to be appended to each event
```

File: README.md [hunk 4]
```diff
@@ -69,3 +60,44 @@ public final class ResolveFlags {
...
+    final FeatureProvider provider = new ConfidenceFeatureProvider(confidence);
+    // Flags are operated via the same OpenFeature Client APIs
+    OpenFeatureAPI.getInstance().setProvider(provider);
+
+    // Additionally, events can be emitted
+    confidence.send("my-event", ConfidenceValue.of("event-value"));
+  }
+}
+```
+The `send()` API currently supports:
```
Comment (nicklasl) [inline README.md:93]:
```suggestion
### The `send()` API currently supports
```

File: README.md [hunk 5]
```diff
@@ -69,3 +60,44 @@ public final class ResolveFlags {
...
+- Setting a custom event's payload with all the Confidence-supported types via the `ConfidenceValue` interfaces
+- Automatically including the OpenFeature's `Evaluation Context` detected at the time `send()` is called
+  - _Note: this only considers the global Evaluation Context set at the OpenFeatureAPI level_
+### Setting a shared `Event Context` that is going to be appended to each event
+
+```java
+    final Confidence confidenceWithContext = confidence.withContext(ConfidenceValue.of("context-value"));
+    confidenceWithContext.send("my-event", ConfidenceValue.of("event-value"));
+```
+The "my-event" event in the example above will contain fields for "event-value", "context-value" and the Evaluation Context data set via `OpenFeatureAPI.getInstance().setEvaluationContext(...)`.
```
Comment (nicklasl) [inline README.md:103]: nit: 
```suggestion
The "my-event" event in the example above will contain fields for "event-value", "context-value" and the Evaluation Context data set via `OpenFeatureAPI.getInstance().setEvaluationContext(...)`.

We suggest you you this `withContext()`-API to send events which are "request context" based events as well.

```

**Review by nicklasl — APPROVED** [review]

File: README.md (hunk 5 above)
Comment (fabriziodemaria) [reply README.md:109]: > We suggest you you this `withContext()`-API to send events which are "request context" based events as well.

Can you re-phrase slightly or explain here what you mean?

**Pull request closed without merging**